wgpu/__pyinstaller/hook-wgpu.py

```diff
@@ -1,17 +1,35 @@
 from PyInstaller.utils.hooks import collect_data_files, collect_dynamic_libs
 
-# Include our data and binaires
-datas = collect_data_files("wgpu")
-binaries = collect_dynamic_libs("wgpu")
-hiddenimports = ["wgpu.resources"]
+# Init variables that PyInstaller will pick up.
+hiddenimports = []
+datas = []
+binaries = []
 
-# Include backends that we want to Just Work
+# Include the resources subpackage and its contents.
+hiddenimports += ["wgpu.resources"]
+datas += collect_data_files("wgpu", subdir="resources")
+binaries += collect_dynamic_libs("wgpu")
+
+# Include backends. Make sure all our backend code is present,
+# and let PyInstaller resolve imports/dependencies for some.
+datas += collect_data_files(
+    "wgpu", subdir="backends", include_py_files=True, excludes=["__pycache__"]
```

Comment:
include_py_files=True ? That seems unnecessary?

Reply:
This way, all `.py` files are included, so that wgpy is at least complete, as a library. E.g. if a downstream lib dynamically imports a gui backend, it'll still work (provided the dependencies are fulfilled).

Reply:
I understand that but it's not beneficial to launch time of the distributable. Pyinstaller packages all the python code in compiled form in a binary format that can be loaded in a single disk operation. Including python files as data files is not going to benefit from that.

Application builders will have to statically import gui backends (this covers majority of cases I think) or add a hiddenimports line to their pyinstaller spec file. This is standard practise for pyinstaller, so I don't think we should deviate from that.

Reply:
Ok, I guess I was trying to hard. I will simplify the hook accordingly.